src/module/api/timeline/create/index.ts

```diff
@@ -26,8 +26,10 @@ export async function Create(newTimeline: INewTimeline): Promise<any> {
 
   console.log(req.toObject());
 
+  const metadata = { Authorization: "Bearer <token>" };
+
   const getCreateResponsePb = await new Promise((resolve, reject) => {
-    client.create(req, {}, function (err: any, res: CreateO) {
+    client.create(req, metadata, function (err: any, res: CreateO) {
```

Comment:
If I got it right this is what we have to do in all api calls now. We need to provide an **access token** in the metadata of the gRPC call. Here with emulate the HTTP authorization header and use the bearer scheme. The backend knows then how to deal with that and will reject requests not providing a valid JWT.

Reply:
Love it. We're turning into a real app!

Reply:
Can you check the links below? We can implement an interceptor just like on the server side. The interceptor is configured for the client so that we do not need to put the metadata for every RPC. 

- https://grpc.io/blog/grpc-web-interceptor
- https://nicu.dev/posts/typescript-grpc-web-auth-interceptor

Reply:
I'm having trouble understanding the benefit of using an interceptor in the web-client side. It could also be I'm implementing calls incorrectly...

It seems like a more complex way of implementing your original suggestion:

![Learn_how_gRPC_interceptors_works_in_Golang_and_React___ITNEXT](https://user-images.githubusercontent.com/1753324/108798998-0bbb9200-7544-11eb-9771-e65184ffe208.jpg)

links in article:
- https://github.com/grpc/grpc-web/blob/master/net/grpc/gateway/examples/echo/commonjs-example/client.js#L36
- https://github.com/grpc/grpc-web/issues/766
- https://github.com/grpc/grpc-web/issues/942

Reply:
The basic idea of interceptors is to configure a client with an interceptor. The interceptor can then intercept each call. In our case we would want to intercept each call using the interceptor in order to set the authorization header in the metadata. The benefit would be to configure the interceptor once for the client, instead of configuring the metadata for each call. The interceptor is meant to make things more simple, if this is not the case we are doing it wrong or should not do it at all in Javascript. In Golang it works well. 

<img width="741" alt="Screenshot 2021-02-23 at 11 35 32" src="https://user-images.githubusercontent.com/552769/108832056-4288bb00-75cb-11eb-9bd8-34361b4868ed.png">